src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -572,23 +575,199 @@ void drawing_classifier::train(gl_sframe data, std::string target_column_name,
         validation_data_, target_column_name, "report", val_predictions,
         {{"classes", read_state<flex_list>("classes")}});
 
-    for (auto &p : val_metric) {
+    for (auto& p : val_metric) {
       state_update["validation_" + p.first] = p.second;
     }
   }
 
   add_or_update_state(state_update);
 }
 
+gl_sframe drawing_classifier::perform_inference(data_iterator* data) const {
+  // Open a new SFrame for writing.
+  gl_sframe_writer writer({"preds"}, {flex_type_enum::VECTOR},
```

Comment:
Can we have comment annotations for the other arguments as well?

Reply:
no. The code says itself.

Reply:
What is `{flex_type_enum::VECTOR}`?

Reply:
The data type for column "preds".